File: leetcode/1201-1500/1451-1500/1480.Running Sum of 1d Array.cc
// https://leetcode.com/problems/running-sum-of-1d-array/submissions/

static int x=[](){
  std::ios::sync_with_stdio(false);
  cin.tie(nullptr);
  cout.tie(nullptr);
  return 0;
}();

class Solution {
public:
    vector<int> runningSum(vector<int>& nums) {
        for(unsigned i = 1; i < nums.size(); ++i) 
        {
            nums[i] += nums[i - 1];
        }
        return move(nums);
    }
};

File: leetcode/1201-1500/1451-1500/1480.Running Sum of 1d Array.rs
impl Solution {
    pub fn running_sum(mut nums: Vec<i32>) -> Vec<i32> {
        let mut result = Vec::with_capacity(nums.len());
        result.push(nums[0]);
        for idx in 1..nums.len() {
            result.push(nums[idx] + result[idx - 1]);
        }
        result
    }
}